### PR TITLE
feat: config check is Config's duty.

### PR DESCRIPTION
**Pull request by ForrestSu**

### CODE REVIEW

File: iterator.go [hunk 1]
```diff
@@ -13,9 +13,6 @@ func (e iteratorError) Error() string {
 // ErrInvalidIteratorState is reported when iterator is in invalid state
 const ErrInvalidIteratorState = iteratorError("Iterator is in invalid state. Use SetNext() to move to next position")
 
-// ErrCannotRetrieveEntry is reported when entry cannot be retrieved from underlying
-const ErrCannotRetrieveEntry = iteratorError("Could not retrieve entry from cache")
```
Comment (cristaloleg) [inline iterator.go:17]: This breaks backward compatibility

Comment (ForrestSu) [reply iterator.go:17]: This err doesn't seem to be used anymore, to keep the code clean, can it be removed directly？

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# BigCache [![Build Status](https://github.com/allegro/bigcache/workflows/build/badge.svg)](https://github.com/allegro/bigcache/actions?query=workflow%3Abuild)&nbsp;[![Coverage Status](https://coveralls.io/repos/github/allegro/bigcache/badge.svg?branch=master)](https://coveralls.io/github/allegro/bigcache?branch=master)&nbsp;[![GoDoc](https://godoc.org/github.com/allegro/bigcache/v3?status.svg)](https://godoc.org/github.com/allegro/bigcache/v3)&nbsp;[![Go Report Card](https://goreportcard.com/badge/github.com/allegro/bigcache/v3)](https://goreportcard.com/report/github.com/allegro/bigcache/v3)
```
[the remark this answers is not in the document]
Comment (ForrestSu) [reply README.md:1]: ok

[the remark this answers is not in the document]
Comment (ForrestSu) [reply README.md]: > Can you do it in a separated PR?

fixed

**Pull request closed without merging**